lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -238,5 +239,50 @@
 
         return a;
     });
-  
+
+    platform.registerFunction(FunctionsEnum.ArrayForEach, function (callbackfn, thisArg) {
+        // ECMAScript 2017 #sec-array.prototype.foreach
+        "use strict";
+
+        if (this === null || this === undefined) {
+            __chakraLibrary.raiseThis_NullOrUndefined("Array.prototype.forEach");
+        }
+
+        let o;
+        let len;
+        if (__chakraLibrary.isArray(this)) {
+            o = this;
+            len = o.length;
+        } else {
+            o = __chakraLibrary.Object(this);
+            len = __chakraLibrary.GetLength(o);
+        }
+
+        if (typeof callbackfn != "function") {
+            __chakraLibrary.raiseFunctionArgument_NeedFunction("Array.prototype.forEach");
```

Comment:
could this boilerplate code be refactored with other functions? I assume there will be other implementation in the future which can make use of that.

Reply:
The code have been refactored